spec/context.md

```diff
@@ -0,0 +1,392 @@
+# About this file
+
+This is an attempt to document the context-0.2 compression format.
+
+WARNING: the content is almost just a copy of entropy.md and doesn't reflect
+the actual format.  They're about to be changed so much once we fixed the
+format.  The current content is here to clarify the difference between entropy
+in the future changes.
+
+# Global structure
+
+```
+Stream ::= Headers Prelude? Content? Main Footers?
+```
+
+# Headers
+
+Headers identify the file, the version of the format used and the shared dictionary used, if any.
+
+The shared dictionary provides:
+
+- a family of context-based prediction tables;
+- well-known literal strings;
+- well-known floating-point numbers;
+- well-known unsigned long numbers;
+- well-known property keys;
+- well-known identifier names;
+- well-known list lengths.
+
+```
+Headers ::= MagicNumber FormatVersionNumber LinkToSharedDictionary
+```
+
+## Magic number
+
+```
+MagicNumber ::= "\x89BJS\r\n\0\n"
+```
+
+
+## Version number
+
+Not specified/implemented yet.
+
+```
+VersionNumber ::= TBD
+```
+
+## Link to shared dictionary
+
+Not specified/implemented yet.
+
+The shared dictionary is optional.
+
+```
+LinkToSharedDictionary ::= TBD
+```
+
+
+# Prelude
+
+The **prelude** extends the shared dictionary with additional strings, numbers, keys, names, ...
+
+Data structures are separated even when they have the same underlying representation, to improve
+compression.
+
+```
+Prelude ::= "PREL" NamedPreludeDictionary*
+NamedPreludeDictionary ::= "FLOA" PreludeStream<f64?>
+                         | "ULON" PreludeStream<u32?>
+                         | "LIST" PreludeStream<u32?>
+                         | "PROP" PreludeStream<String>
+                         | "IDEN" PreludeStream<String>
+                         | "STRI" PreludeStream<String>
+                         | "PROp" PreludeStream<StringLen>
```

Comment:
`PROP` vs. `PROp` is not very readable.

Maybe `PROP` vs. `prop`? `PROC` vs. `PROL` (`C` = content, `L` = len)?

Reply:
thanks. I'll use C/L